### PR TITLE
Add OPA document editor formatter

### LINKED ISSUE #36: Add OPA document formatter
Register OPA DocumentFormattingEditProvider for rego filetype using `opa fmt` CLI

### PR DESCRIPTION
**Pull request by rcbop**
Register OPA DocumentFormattingEditProvider for rego filetype using `opa fmt` CLI

Closes #36

### CODE REVIEW

**Comment by rcbop** [comment]
Closes Checkmarx/kics#1800

**Review by srenatus — APPROVED** [review]
🎉 What a nice contribution, thank you! The code looks OK as far as I can tell, but I'm not an expert here... (Haven't pulled it down and run it yet either.)

File: README.md [hunk 1]
```diff
@@ -76,6 +77,6 @@ If you want to hack on the extension itself, you should clone this repository, i
 
 ## ROADMAP
 
-* [ ] run `opa fmt` on save to reformat file
+* [x] run `opa fmt` on save to reformat file
```
Comment (tsandall) [inline README.md:80]: Nit: Just remove this from the list and add a bullet at the top to indicate it's supported.

**Review by tsandall — APPROVED** [review]
LGTM -- just one minor nit on the README.

File: README.md (hunk 1 above)
Comment (rcbop) [reply README.md]: done!

**Comment by tsandall** [comment]
Thanks @rcbop !

**Pull request merged**